programs/KJVDataDump/main.cpp

```diff
@@ -371,14 +396,44 @@ int main(int argc, char *argv[])
 						if (bOutputTemplates) {
 							std::cout << pVerse->m_strTemplate.toUtf8().data() << std::endl;
 						}
-						if (bOutputVerseText) {
-							if (bOutputTemplates) std::cout << strSpacer.toUtf8().data();
-							if (bOutputTransChangeAdded) {
-								std::cout << lstVerseWords.join(QChar(' ')).toUtf8().data();
-							} else {
-								std::cout << strParsedVerse.toUtf8().data();
+						if (bOutputVerseText && nLetterCount > 0) {
+							QString verse = strParsedVerse.toUtf8().data();
+							QString clean = QString(verse).toLower().simplified().replace(QRegularExpression("[\\[\\]()\.,:;!?]"), "");
```

Comment:
Instead of starting with richified verse text and stripping it all back down, it would be better to process it in the VerseTextRichifier itself as the verse text is being generated.

You can create your own `CVerseTextRichifier` class here in this code, derived from `CVerseTextRichifier` and override the `pushWordToVerseText` member.  That member is specifically declared as virtual to allow derived classes to alter how the word being rendered, but it also let's your code "see" the each word as it's rendered.  And, it lets you access database flags about the word, like if it's a translation add/change [i.e. italics] word or if it's a red-letter "Words of Jesus" word.

Your derived class could compute the letter counts for the words as the verse is being synthesized, rather than trying to decompose it back to bare words and undo what the richifier already did.

See this function in the base CVerseTextRichifier class:
https://github.com/dewhisna/KingJamesPureBibleSearch/blob/23b304fad00b5b8b8a60c07c887d2dbe3df6723a/programs/KJVCanOpener/VerseRichifier.cpp#L310-L342

Your class can use this to "see" the word and then just call the base class implementation to continue with normal verse rendering.